portfolio/src/main/webapp/script.js

```diff
@@ -38,18 +37,121 @@ function showModal(img) {
-    console.log(greeting);
-    document.getElementById('data-container').innerText = greeting;
+
+  // Get user's values from the html document
+  const maxElement = document.getElementById("quantity");
+  const maxToDisplay = maxElement.value;
+  const sort = document.getElementById("sort").value;
+  const imgsrc = document.getElementById("modal-img").src;
+  const img = getImgName(imgsrc);
+  
+  fetch("/comment?quantity=" + maxToDisplay + "&sort=" + sort + "&img=" + img).then(response => response.json()).then(jsonObj => {
+    const comments = jsonObj.comments;
+    const total = jsonObj.total;
+
+    // Display the total number of comments to user
+    const totalElement = document.getElementById("total");
+    totalElement.innerText = ("Comments (" + total + ")");
+
+    // Make sure user can't try to display more than the total number of comments
+    maxElement.max = total.toString();
+    maxElement.placeholder = total.toString();
+
+    // Add individual comments to html page
+    const list = document.getElementById("all-comments");
+    list.innerHTML = "";
+    comments.forEach((comment) => {
+      const listElement = document.createElement("li");
+      
+      // Display name, date, and comment
+      const textElement = document.createElement("span");
+      const date = new Date(comment.timestamp);
+      textElement.innerHTML = "<b>" + comment.name + "</b>    <small>" + date.toLocaleString() + "</small><br><br>" + comment.comment;
+
+      // Display a delete button
+      const deleteButtonElement = document.createElement("button");
+      deleteButtonElement.className = "delete-button";
+      deleteButtonElement.innerText = "Delete";
+      deleteButtonElement.addEventListener("click", () => {
+        console.log("Delete button clicked");
+        deleteComment(comment);
+      });
+      
+      listElement.appendChild(textElement);
+      listElement.appendChild(deleteButtonElement);
+      list.append(listElement);
+    })
   });
+}
+
+// Create POST request to /delete servlet to delete comment, called by delete button onclick
+function deleteComment(comment) {
+  console.log("Deleting comment");
+
+  const params = new URLSearchParams();
+  params.append("id", comment.id);
+  const postRequest = new Request("/delete", {method: "POST", body: params});
+  fetch(postRequest).then(() => getServerData()).catch(error => console.error(error));
+}
+
+// Create POST request to /comment servlet
+function postComment() {
+  console.log("Posting comment");
+
+  // Get user's values 
+  const nameElement = document.getElementById("name");
+  const commentElement = document.getElementById("comment-box");
+  const imgsrc = document.getElementById("modal-img").src;
+
+  // Don't create comment if the comment is empty or the image has no src
+  if (!commentElement.value || !imgsrc) {
+    console.log("null comment or image src");
+    return;
+  }
+
+  // Get image name that comment is under
+  const img = getImgName(imgsrc);
```

Comment:
I assume at this step img could still be "" if imgsrc is somehow not valid?

Reply:
you're right. I forgot to add the check in my get method